File: p4_suma.py
import psycopg2
from psycopg2 import sql

# Conexión con la base de datos
conn = psycopg2.connect(
    dbname='corto1', 
    host='localhost', 
    port='5432', 
    user='postgres', 
    password='523811')

# Crear un cursor para ejecutar consultas
cur = conn.cursor()

# Solicitar un número
num = int(input('Ingrese un número: '))

# Calcular la suma de los números desde 0 hasta el número ingresado
sum = 0
for i in range(num + 1):
    sum += i

# Mostrar la suma en la consola
print(f"La suma de los números desde 0 hasta {num} es: {sum}")

try: 
    # Insertar la información en la tabla "problema4"
    query = sql.SQL("INSERT INTO problema4 (num, sum, operacion) VALUES (%s, %s, %s);")
    cur.execute(query, (num, sum, 'Suma'))
    conn.commit()

except (Exception, psycopg2.DatabaseError) as error:
    print(f"Error durante la conexión a la DB, consulte sobre el error: {error}")

# Cerrar la conexión con la base de datos
conn.close()

File: p4_suma.m
%PROBLEMA 4 -- suma de números
%Conexión con la base de datos
pkg load database
conn=pq_connect(setdbopts('dbname','corto1','host','localhost','port','5432','user','postgres','password','523811'))

% Crear tabla en base de datos
%tablename = 'problema4';
%query = ['CREATE TABLE ', tablename, ' (num INT, sum INT, operacion TEXT);'];
%pq_exec_params(conn, query);


% Solicitar un número
num = input('Ingrese un número: ');

% Calcular la suma de los números desde 0 hasta el número ingresado
sum = 0;
for i = 0:num
    sum = sum + i;
end

% Mostrar la suma en la consola
disp(['La suma de los números desde 0 hasta ', num2str(num), ' es: ', num2str(sum)]);

% Insertar la información en la tabla "problema4"
query = ['INSERT INTO ', tablename, ' (num, sum, operacion) VALUES (', num2str(num), ', ', num2str(sum), ', ''Suma'');'];
pq_exec_params(conn, query);

% Cerrar la conexión con la base de datos
pq_close(conn);
